### ISSUE TITLE
Investigate issue with data download link - WA Acoustic Wave

### ISSUE THREAD

**Issue by AnaBelgun** [TerriaJS/nationalmap#833]
WA Gov Transport GIS team pointed out that when trying to download the following data (through the link provided as a feature, and available in the feature info panel), users get an errror: 

https://www.nationalmap.gov.au/#share=s-cyncQO6djOvwIE15xpaNjrBIc4X

![image](https://user-images.githubusercontent.com/18452161/53130022-3ec99480-35bd-11e9-898c-789ffff1c375.png)
 When you click on Binary Files or Text Files links (https://www.nationalmap.gov.au/%22https://s3-ap-southeast-2.amazonaws.com/transport.wa/DOT_OCEANOGRAPHIC_SERVICES/AWAC_V2/KALBARRI_AWAC/KAL01_Binary.zip%22) users get a 404 error:

![image](https://user-images.githubusercontent.com/18452161/53130064-61f44400-35bd-11e9-82ea-22b8d72ddfaa.png)

**Comment by AnaBelgun** [comment]
I've checked the actual resource link http://catalogue.beta.data.wa.gov.au/dataset/awac-acoustic-wave-and-current-stations on data.wa.gov.au and when trying to download the data, users are prompted to log in: 
![image](https://user-images.githubusercontent.com/18452161/53130217-bac3dc80-35bd-11e9-9f7a-656a81410ad9.png)

When clicking on "go  to resource", users get  "error - access denied"

![image](https://user-images.githubusercontent.com/18452161/53130292-eb0b7b00-35bd-11e9-99d1-de2254c8274c.png)

**Comment by meh9** [comment]
There are some issues here with quoting, but I'm not sure if the problem is on TerriaJS's side or on the data custodian side.

The data being sent in response to the feature info request looks to me to be properly HTML escaped so as not to interfere with the `text/xml` return type, but for some reason when we unescape it we are not treating the links as absolute any more, and are in fact including the `""` marks in the URL.

Here's a snippet of the response XML:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<FeatureInfoResponse xmlns:esri_wms="http://www.esri.com/wms" xmlns="http://www.esri.com/wms">
<FIELDS objectid="13" SymbolID="0" Snippet="09/08/2006 - 20/11/2006" 
  PopupInfo="&lt;img src=&amp;quot;http://www.transport.wa.gov.au/images-system/header-logo.png&amp;quot;&gt;&#xA;&lt;/p&gt;&#xA;&lt;div class='googeb-info-window' style='font-family: arial'&gt;&#xA;&lt;b&gt;AWAC - Acoustic Wave and Current Stations&lt;/b&gt;&lt;br&
```

@kring any thoughts on where the problem lies?

**Comment by kring** [comment]
The problem here is the the URLs are double escape in the XML returned by the server for the `GetFeatureInfo` service. Here's an example response:

```
<?xml version="1.0" encoding="UTF-8"?>

<FeatureInfoResponse xmlns:esri_wms="http://www.esri.com/wms" xmlns="http://www.esri.com/wms">
<FIELDS objectid="13" SymbolID="0" Snippet="09/08/2006 - 20/11/2006" PopupInfo="&lt;img src=&amp;quot;http://www.transport.wa.gov.au/images-system/header-logo.png&amp;quot;&gt;&#xA;&lt;/p&gt;&#xA;&lt;div class='googeb-info-window' style='font-family: arial'&gt;&#xA;&lt;b&gt;AWAC - Acoustic Wave and Current Stations&lt;/b&gt;&lt;br&gt;&#xA;&lt;/div&gt;&#xA;&lt;hr&gt;&#xA;&lt;div class='googeb-info-window' style='font-family: sans-serif'&gt;&#xA;&lt;b&gt;STATION INFORMATION&lt;/b&gt;&lt;br&gt;&#xA;&lt;br&gt;&#xA;&lt;b&gt;AREA LOCATION NAME:&lt;/b&gt; KALBARRI&lt;br&gt;&#xA;&lt;b&gt;AWAC LOCATION DESCRIPTION:&lt;/b&gt;  KALBARRI ENTRANCE&lt;br&gt;&#xA;&lt;b&gt;AWAC LOCATION ID:&lt;/b&gt;  KAL01&lt;br&gt;&#xA;&lt;b&gt;AWAC DATA AVAILABILITY:&lt;/b&gt;  09/08/2006 TO  20/11/2006&lt;br&gt;&#xA;Note: Data may not be continuous, please refer to the metadata file in the data folders for exact deployment duration.&lt;br&gt;&#xA;&lt;br&gt;&#xA;&lt;b&gt;LATITUDE:&lt;/b&gt;  -27.708683&lt;br&gt;&#xA;&lt;b&gt;LONGITUDE:&lt;/b&gt;  114.160067&lt;br&gt;&#xA;&lt;b&gt;DEPTH&lt;/b&gt;:  3.5(Metres)&lt;br&gt;&#xA;&lt;hr&gt;&#xA;&lt;b&gt;AWAC DATA DESCRIPTIONS&lt;/b&gt;&lt;br&gt;&#xA;&lt;br&gt;&#xA;&lt;a href=&amp;quot;https://s3-ap-southeast-2.amazonaws.com/transport.wa/DOT_OCEANOGRAPHIC_SERVICES/AWAC/AWAC_READ_ME/AWAC_READ_ME.htm&amp;quot;target=&amp;quot;_blank&amp;quot;&gt;Acoustic Wave and Current ReadMe&lt;/a&gt;&lt;br&gt;&#xA;&lt;a href=&amp;quot;https://s3-ap-southeast-2.amazonaws.com/transport.wa/DOT_OCEANOGRAPHIC_SERVICES/AWAC/AWAC_READ_ME/AWAC_SUPPORTING_INFORMATION.zip&amp;quot;target=&amp;quot;_blank&amp;quot;&gt;Additional ReadMe Information&lt;/a&gt;&lt;br&gt;&#xA;&lt;br&gt;&#xA;Note: Please refer to READ ME files for information about the data files and data format.&#xA;&lt;br&gt;&#xA;&lt;hr&gt;&#xA;&lt;b&gt;ACOUSTIC WAVE AND CURRENT DATA&lt;/b&gt;&lt;br&gt;&#xA;&lt;br&gt;&#xA;&lt;a href= &amp;quot;https://s3-ap-southeast-2.amazonaws.com/transport.wa/DOT_OCEANOGRAPHIC_SERVICES/AWAC_V2/KALBARRI_AWAC/KAL01_Text.zip&amp;quot; target=&amp;quot;_blank&amp;quot;&gt;Text Files For KALBARRI&lt;/a&gt;&lt;br&gt;&#xA;&lt;br&gt;&#xA;&lt;a href= &amp;quot;https://s3-ap-southeast-2.amazonaws.com/transport.wa/DOT_OCEANOGRAPHIC_SERVICES/AWAC_V2/KALBARRI_AWAC/KAL01_Binary.zip&amp;quot; target=&amp;quot;_blank&amp;quot;&gt;Binary Files For KALBARRI&lt;/a&gt;&lt;br&gt;&#xA;Warning: Raw binary files can be really big and take long time to download.&lt;br&gt;&#xA;&lt;/div&gt;&#xA;&lt;hr&gt;&#xA;&lt;a href=&amp;quot;http://creativecommons.org/licenses/by/4.0/&amp;quot; rel=&amp;quot;license&amp;quot; target=&amp;quot;_blank&amp;quot;&gt;&lt;img alt=&amp;quot;Creative Commons Licence&amp;quot; src=&amp;quot;http://i.creativecommons.org/l/by/4.0/88x31.png&amp;quot; style=&amp;quot;border-width:0&amp;quot;/&gt;&lt;/a&gt;&lt;br/&gt;This work is licensed under a &lt;a href=&amp;quot;http://creativecommons.org/licenses/by/4.0/&amp;quot; rel=&amp;quot;license&amp;quot; target=&amp;quot;_blank&amp;quot;&gt;Creative Commons Attribution 4.0 International License&lt;/a&gt;.&lt;/hr&gt;&#xA;&lt;hr&gt;&#xA;Dataset Produced on 19/12/2017 16:35&#xA;&lt;hr&gt;" Name="KALBARRI" description="AWAC - Acoustic Wave and Current Stations" Shape="Point"></FIELDS>
</FeatureInfoResponse>
```

The important bit is:
```
PopupInfo="&lt;img src=&amp;quot;http://www.transport.wa.gov.au ...
```

The `&lt;` is an XML entity reference. When we load the XML, it is dutifully decoded to `<`. Similarly, `&amp;` is decoded to `&`. So, after decoding the value of that `PopupInfo` attribute, we get

```
<img src=&quot;http://www.transport.wa.gov.au ...
```

But wait! That `&quot;` is another entity reference. Too bad. It won't be decoded because it didn't exist in the first pass. The `&` at the start was escaped by replacing it with `&amp;`. In other words, the `src` attribute was incorrectly escaped _twice_.

The server should have returned this:

```
PopupInfo="&lt;img src=&quot;http://www.transport.wa.gov.au ...
```

In which case Terria (the web browser's built-in XML parser, really) would correctly unescape it on load and create a link to the correct URL. But because it was escaped incorrectly, the URL gets mangled, and the generated link is broken. We can't fix this in TerriaJS without undesirable side-effects, like preventing the use of _intentionally_ escaped elements.

TL;DR: the server needs to be fixed to return a correctly-escaped `PopupInfo` field.

**Comment by AnaBelgun** [comment]
emailed data custodian with the response

**Comment by AnaBelgun** [comment]
From WA team: Hi, Thanks for the feedback, I’ll have this addressed by the SLIP team. Regards
